lib/rubocop/cop/style/open_ssl_digest_constant.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Algorithmic constants for OpenSSL::Digest are soft-deprecated since
+      # version 2.2.0. Prefer passing a string instead to use hash functions
+      # without defined constants like SHA512-256, SHA3-256 or BLAKE2s256.
+      #
+      # @example instantiation
+      #
+      #   # bad
+      #   OpenSSL::Digest::SHA256.new
+      #
+      #   # good
+      #   OpenSSL::Digest.new('SHA256')
+      #
+      # @example ::Digest inherited class methods
+      #
+      #   # bad
+      #   OpenSSL::Digest::SHA256.digest('foo')
+      #
+      #   # good
+      #   OpenSSL::Digest.digest('SHA256', 'foo')
+      #
+      class OpenSSLDigestConstant < Cop
+        INIT_MSG = "Use `OpenSSL::Digest.%<method>s('%<algorithm>s')` " \
+          'instead of `%<original>s`.'
+        WITH_ARGS_MSG = "Use `OpenSSL::Digest.%<method>s('%<algorithm>s', " \
+          '%<args>s)` instead of `%<original>s`.'
+
+        def_node_matcher :digest_const, <<~PATTERN
+          (send
+            $(const
+              (const
+                (const {nil? cbase} :OpenSSL) :Digest)
+              _)
+            ...)
+        PATTERN
+
+        def on_send(node)
+          add_offense(node) if digest_const(node)
+        end
+
+        # rubocop:disable Metrics/AbcSize
+        def autocorrect(node)
+          digest_constant, = digest_const(node)
+          algorithm = algorithm_name(digest_constant)
+
+          lambda do |corrector|
+            corrector.remove(digest_constant.loc.double_colon)
+            corrector.remove(digest_constant.loc.name)
+
+            if digest_initialisation?(node)
+              corrector.insert_after(node.loc.selector, "('#{algorithm}')")
+            else
+              corrector.insert_after(node.loc.begin, "'#{algorithm}', ")
+            end
+          end
+        end
+        # rubocop:enable Metrics/AbcSize
+
+        private
+
+        def message(node)
+          digest_constant, = digest_const(node)
+          algorithm = algorithm_name(digest_constant)
+          method = node.loc.selector.source
+
+          if digest_initialisation?(node)
+            format(INIT_MSG,
+                   method: method, algorithm: algorithm, original: node.source)
+          else
+            args = node.children.last.source
+            format(WITH_ARGS_MSG,
+                   method: method, algorithm: algorithm, original: node.source,
+                   args: args)
+          end
+        end
+
+        def digest_initialisation?(node)
+          node.loc.selector.source == 'new'
```

Comment:
Can you use `method?`?

```suggestion
          node.method?(:new)
```